dissect/target/plugins/apps/browser/chromium.py

```diff
@@ -248,32 +277,28 @@ def cookies(self, browser_name: str | None = None) -> Iterator[BrowserCookieReco
                         local_state_parent = local_state_parent.parent
                     local_state_path = local_state_parent.joinpath("Local State")
 
-                    decrypted_key = self._get_local_state_key(local_state_path, user.user.name)
-                except ValueError:
-                    self.target.log.warning("Failed to decrypt local state key")
+                    keys = self.decryption_keys(local_state_path, user.user.name)
+                except ValueError as e:
+                    self.target.log.warning("Failed to decrypt local state key in %s: %s", local_state_path, e)
+                    self.target.log.debug("", exc_info=e)
 
             try:
                 for cookie in db.table("cookies").rows():
                     cookie_value = cookie.value
 
-                    if (
-                        not cookie_value
-                        and decrypted_key
-                        and (enc_value := cookie.get("encrypted_value"))
-                        and enc_value.startswith(b"v10")
-                    ):
+                    if not cookie_value and keys and (encrypted_cookie := cookie.get("encrypted_value")):
+                        callable = DECRYPT_MAP.get(self.target.os, {}).get(encrypted_cookie[0:3], decrypt_unsupported)
```

Comment:
The pattern for using `DECRYPT_MAP` looks always the same (correct me if I'm wrong), get callable and immediately call. I'd just replace it with a function `result = decrypt_value(target, user, keys, value)`.

You can still have that `DECRYPT_MAP` dict in the `decrypt_value` function, but just inlined in the body of it. When we drop Python 3.9 (which will be soon-ish, this summer), we could replace it with a match pattern: https://docs.python.org/3/whatsnew/3.10.html#pep-634-structural-pattern-matching